addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -324,8 +324,20 @@ var Wysiwyg = Widget.extend({
             await this.snippetsMenu.appendTo($mainSidebar);
 
             // Place the history buttons in their right location.
-            $('.o_we_website_top_actions .o_we_external_history_buttons').append($('button[data-action=undo]'));
-            $('.o_we_website_top_actions .o_we_external_history_buttons').append($('button[data-action=redo]'));
+            const $undoButton = $('<button name="undo" class="btn btn-secondary fa fa-undo"></button>');
+            const $redoButton = $('<button name="redo" class="btn btn-secondary fa fa-repeat"></button>');
+            $undoButton.on('click', () => this.editor.execCommand('undo'));
+            $redoButton.on('click', () => this.editor.execCommand('redo'));
+            const reactiveEditorInfo = this.editor.plugins.get(JWEditorLib.ReactiveEditorInfo);
+            const updateButtons = () => {
+                const state = reactiveEditorInfo.editorInfo.get();
+                state.canUndo ? $undoButton.removeAttr('disabled') : $undoButton.attr('disabled', true);
+                state.canRedu ? $redoButton.removeAttr('disabled') : $redoButton.attr('disabled', true);
```

Comment:
You obviously did not test this code before asking me to review it. 👎

Reply:
Why do you say that ?
I've tested it and it works perfectly.

As I said as a comment of this PR, it's gonna work only with that branch: odoo-dev/jabberwock#432

Reply:
It is literally *impossible* for this code to work because your jabberwock PR does not expose a `canRedu` property.